azure/cargo-check.yml

```diff
@@ -33,17 +26,16 @@ jobs:
         echo '##vso[task.setvariable variable=is_locked]false';
       fi
     displayName: Are dependencies locked?
-  # we only check with --all here since we don't care about the actual
-  # compilation, just the dependencies + dev-dependencies resolution.
-  # we _do_ include all features as it may matter though
```

Comment:
This change ignores the contents of this comment. We specifically want to check `--locked` with `--all-features` (unless the user has opted out of it), because otherwise there may be optional dependencies that are brought in by additional features that we don't check against locked. Is there a particular reason you want to change that behavior?

Reply:
Interesting, is that actually the case? I'm pretty sure I've seen errors in the past where optional dependencies were only _resolved_ when they were brought into play by enabling a feature?

Reply:
Confirmed: I took the same setup as mentioned elsewhere but made the dependency optional and `--locked` failed as expected.